File: Source code/src/main/java/vn/edu/hcmut/smartwarehouse/product/Product.java
package vn.edu.hcmut.smartwarehouse.product;

import java.io.Serializable;
import java.time.LocalDateTime;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.edu.hcmut.smartwarehouse.product.type.ProductType;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@Entity
@Table(name = "product")
public class Product implements Serializable {
	private static final long serialVersionUID = -5984908048402920520L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	private String name;

	@ManyToOne(cascade = { CascadeType.PERSIST })
	@JoinColumn(name = "type_product")
	private ProductType type;

	private String image;

	private int weight;

	private int amount;

	private ProductStatus status;

	private LocalDateTime inDate;

	private LocalDateTime outDate;

	private LocalDateTime expiryDate;
}

File: Frontend/src/app/shared/models/product.ts
export interface Product {
  id: number;
  name: string;
  image: string;
  type: ProductType;
  status: ProductStatus;
  weight: number;
  amount: number;
  inDate: Date;
  outDate: Date;
  expiryDate: Date;
}

export interface ProductType {
  id: number;
  name: string;
  unit: string;
}

export enum ProductStatus {
  OUT_DATE = 'OUT_DATE',
  INITIAL = 'INITIAL',
  AVAILABLE = 'AVAILABLE',
  PROCESS = 'PROCESS',
}
